lxd/auth/authorization_objects.go

```diff
@@ -0,0 +1,208 @@
+	case ObjectTypeInstance:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing instance name argument")
+		}
+
+		return ObjectInstance(projectName, args[0]), nil
+	case ObjectTypeNetwork:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing network name argument")
+		}
+
+		return ObjectNetwork(projectName, args[0]), nil
+	case ObjectTypeNetworkACL:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing network ACL name argument")
+		}
+
+		return ObjectNetworkACL(projectName, args[0]), nil
+	case ObjectTypeNetworkZone:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing network zone name argument")
+		}
+
+		return ObjectNetworkZone(projectName, args[0]), nil
+	case ObjectTypeProfile:
+		if len(args) != 1 {
+			return "", fmt.Errorf("Missing profile name argument")
+		}
+
+		return ObjectProfile(projectName, args[0]), nil
+	case ObjectTypeStorageBucket:
+		if len(args) != 2 {
+			return "", fmt.Errorf("Missing required arguments for a storage bucket object")
+		}
+
+		return ObjectStorageBucket(projectName, args[0], args[1]), nil
+	case ObjectTypeStoragePoolVolume:
+		if len(args) != 3 {
+			return "", fmt.Errorf("Missing required arguments for a storage pool volume object")
+		}
+
+		return ObjectStoragePoolVolume(projectName, args[0], args[1], args[2]), nil
+	}
+
+	return "", fmt.Errorf("Unknown object type %q", objectType)
+}
+
+// ObjectFromRequest returns an object created from the request by evaluating the given mux vars.
+// Mux vars must be provided in the order that they are found in the endpoint path. If the object
+// requires a project name, this is taken from the project query parameter unless the URL begins
+// with /1.0/projects.
+func ObjectFromRequest(r *http.Request, objectType ObjectType, muxVars ...string) (string, error) {
+	muxValues := make([]string, len(muxVars))
+	for i, muxVar := range muxVars {
+		muxValue, err := url.PathUnescape(mux.Vars(r)[muxVar])
+		if err != nil {
+			return "", err
+		}
+
+		if muxValue == "" {
+			return "", fmt.Errorf("Mux var %q not found for object type %q", muxVar, objectType)
+		}
+
+		muxValues[i] = muxValue
+	}
+
+	values, err := url.ParseQuery(r.URL.RawQuery)
+	if err != nil {
+		return "", err
+	}
+
+	projectName := "default"
+	if values.Has("project") {
+		projectName = values.Get("project")
+	}
+
+	// If using projects API we want to pass in the mux var, not the query parameter.
+	if objectType == ObjectTypeProject && strings.HasPrefix(r.URL.Path, "/1.0/projects") {
+		if len(muxValues) != 1 {
+			return "", fmt.Errorf("Missing project name mux var")
+		}
+
+		return ObjectProject(muxValues[0]), nil
+	}
+
+	return ObjectByType(objectType, projectName, muxValues...)
+}
+
+// ObjectParts deconstructs an object into an object type and a string slice of components that comprise the object.
+// An error is returned if the object does not have the expected format.
+func ObjectParts(object string) (ObjectType, []string, error) {
```

Comment:
Similar to comment above, im still not really clear from reading this commit what an "object" is in this context, so I feel the comments and/or the type could be clarified here.

Reply:
This has been removed.